File: Lista de exercicio/projetoPYTHON/questao15/questao15.py
class SaldoInsuficienteException(Exception):
    def __init__(self, mensagem="Saldo insuficiente para realizar o saque."):
        self.mensagem = mensagem
        super().__init__(self.mensagem)

class ContaBancaria:
    def __init__(self, titular, saldo_inicial):
        self.__saldo = saldo_inicial
        self.titular = titular

    def depositar(self, quantia):
        self.__saldo += quantia

    def sacar(self, quantia):
        if quantia > self.__saldo:
            raise SaldoInsuficienteException()
        self.__saldo -= quantia

    def exibir_saldo(self):
        return f"Saldo atual: R${self.__saldo}"

def executar_questao15():
    conta = ContaBancaria("Maria", 500)
    
    try:
        conta.sacar(600)
    except SaldoInsuficienteException as e:
        print(e)

    print(conta.exibir_saldo())

File: Lista de exercicio/projetoJAVA/src/projetojava/questao15.java
package projetojava;

class SaldoInsuficienteException extends Exception {
    public SaldoInsuficienteException(String mensagem) {
        super(mensagem);
    }
}

class ContaBancaria {
    private double saldo;
    private String titular;

    public ContaBancaria(String titular, double saldoInicial) {
        this.titular = titular;
        this.saldo = saldoInicial;
    }

    public void depositar(double quantia) {
        saldo += quantia;
    }

    public void sacar(double quantia) throws SaldoInsuficienteException {
        if (quantia > saldo) {
            throw new SaldoInsuficienteException("Saldo insuficiente para realizar o saque.");
        }
        saldo -= quantia;
    }

    public double exibirSaldo() {
        return saldo;
    }
}

public class questao15 {
    public static void main(String[] args) {
        ContaBancaria conta = new ContaBancaria("Maria", 500);

        try {
            conta.sacar(600); 
        } catch (SaldoInsuficienteException e) {
            System.out.println(e.getMessage());
        }

        System.out.println("Saldo atual: R$" + conta.exibirSaldo());
    }
}
